app/models/conversion_host/configurations.rb

```diff
@@ -0,0 +1,70 @@
+module ConversionHost::Configurations
+  extend ActiveSupport::Concern
+  module ClassMethods
+    def notify_configuration_result(op, success, resource_info)
+      Notification.create(
+        :type    => success ? :conversion_host_config_success : :conversion_host_config_failure,
+        :options => {
+          :op_name => op,
+          :op_arg  => resource_info,
+        }
+      )
+    end
+
+    def queue_configuration(op, instance_id, resource, params, auth_user = nil)
+      task_opts = {
+        :action => "Configuring a conversion_host: operation=#{op} resource=(type: #{resource.class.name} id:#{resource.id})",
+        :userid => auth_user
+      }
+      queue_opts = {
+        :class_name  => name,
+        :method_name => op,
+        :instance_id => instance_id,
+        :role        => 'ems_operations',
+        :zone        => resource.ext_management_system.my_zone,
```

Comment:
Is there ever going to be a chance where the ems is nil here? Just wanted to double check, since I think we've bumped into that situation in the past. If so, maybe `.try(:my_zone)`?

Reply:
I can think of a case where `ems` is `nil`: after user pick a Vm or Host and clicking the `enable` button, user deletes its provider. When deleting a provider, MIQ will disconnect the Vm by `niling` its `ems` field.

This racing condition is really slim (likely need a script running to delete the provider while user is clicking the `enable` button about the same time).  Even if this happens, this action has to blow up

Reply:
Ok, then maybe we should add a guard for `resource.ext_management_system` right there, and just bail with a warning if that happens.

Reply:
Now the caller has rescue for this exceptional scenario.